Migrate the Popover component to Floating UI (#1661)
sumup-oss/circuit-ui: .changeset/hot-schools-attack.md

---

> '@sumup/circuit-ui': major

Reviewer: Since this is a major change for Circuit UI v6, we'll point this PR to a new "next" branch, so we don't block the main release channel while we're working on v6.

More docs on the process here: https://circuit.sumup.com/?path=/docs/introduction-contributing-release-process--page

You don't have to do anything, I'll make the new branch soon and update this PR to point to it instead 🙂

Author: Sounds good! Thanks @robinmetral !

Reviewer: Base changed to `next` ✅

---

> Migrate the Popover component from popper to floating-ui. Popper modifiers are longer supported, while offsetProp is exposed to allow for flexible placement of floating element.

Reviewer 2: I believe you have a small typo here, popper modifiers are no longer supported.
Maybe you can rephrase it 
'Migrated the Popover component from Popper to Floating UI. Popper modifiers are no longer supported, use offset prop for flexible placement of floating element instead.

Author: You're right! @amelako. The suggested wording is more correct. I committed the change. Thanks! :)

> Migrated the Popover component from Popper to Floating UI. Popper `modifiers` are longer supported, use `offsetProp` prop for flexible placement of floating element instead. `placement` prop can no longer have `auto*` values. Fixed `ProfileMenu` to use the updated `Popover` API. 

Reviewer 3: It can be helpful to include links to further information for anyone who's curious.

The `ProfileMenu` changes are an implementation detail — since they don't impact developers, I wouldn't mention them.

```suggestion
Migrated the Popover component from [Popper](https://popper.js.org/) to [Floating UI](https://floating-ui.com/). Popper's `modifiers` are longer supported, use the `offset` prop for flexible placement of the floating element instead. The `placement` prop no longer accepts `auto*` values. 
```

Reviewer 3: I'm a perfectionist 🙈

> Migrated the Popover component from [Popper](https://popper.js.org/) to [Floating UI](https://floating-ui.com/). Popper's `modifiers` are longer supported, use the `offset` prop for flexible placement of the floating element instead. The `placement` prop no longer accepts `auto*` values. 

Reviewer 2: ```suggestion
Migrated the Popover component from [Popper](https://popper.js.org/) to [Floating UI](https://floating-ui.com/). Popper's `modifiers` are no longer supported, use the `offset` prop for flexible placement of the floating element instead. The `placement` prop no longer accepts `auto*` values. 
```
